aten/src/ATen/native/cuda/LinearAlgebra.cu

```diff
@@ -492,8 +495,70 @@ void addr_kernel_cuda(TensorIterator &iter, Scalar beta, Scalar alpha) {
   });
 }
 
+// This reduction accumulates results as the type `acc_t`. By default, when
+// `scalar_t` is complex, `acc_t` is the downgraded real number type.
+// Otherwise, `acc_t` and `scalar_t` are the same type.
+template <typename scalar_t, typename acc_t=typename scalar_value_type<scalar_t>::type, typename out_t=typename scalar_value_type<scalar_t>::type>
+void linalg_vector_norm_kernel_cuda_impl(TensorIterator& iter, Scalar val) {
+  double ord_val;
+  if (val.isIntegral(/*includeBool=*/false)) {
+     ord_val = val.to<int64_t>();
+  } else if (val.isFloatingPoint()) {
+     ord_val = val.to<double>();
+  } else {
+     TORCH_CHECK(false, "linalg_vector_norm_kernel_cuda_impl expects norm to be integer or float");
```

Comment:
If this can be triggered by a user then the function name should be the user-facing name "vector_norm" instead of the internal implementation name. 

The word "norm" here should match the documentation's name for the argument, too ("ord", right?)